draft-ietf-httpbis-resumable-upload.md

```diff
@@ -202,6 +202,8 @@ Once the upload resource is available, the target resource MAY send an informati
 
 The server MUST send the `Upload-Offset` header field in the response if it considers the upload active, either when the response is a success (e.g. `201 (Created)`), or when the response is a failure (e.g. `409 (Conflict)`). The `Upload-Offset` field value MUST be equal to the end offset of the entire upload, or the begin offset of the next chunk if the upload is still incomplete. The client SHOULD consider the upload failed if the response has a status code that indicates a success but the offset indicated in the `Upload-Offset` field value does not equal the total of begin offset plus the number of bytes uploaded in the request.
 
+While the content is received, the target resource MAY repeatedly inform the client about the upload progress by sending an informational response with a `104 (Upload Resumption Supported)` status code. In this informational response, the `Upload-Offset` header field MUST be set to the current upload offset. Later offset retrievals ({{offset-retrieving}}) MUST NOT receive an upload offset that is less than the offset reported in the latest informational response, allowing the client to free associated resources.
```

Comment:
To me, the text seems a bit vague regarding if it mandates the inclusion of upload-offset header in the first 104 response.

My preference would go to mandate inclusion of `upload-offset` header field in all 104 responses (even with the first one, which would typically carry a value of `0`), or say that use of that header is optional in all 104 responses.

Maybe it would make sense to go with the latter, considering that `upload-offset` is an optimization that we consider optional.

Reply:
Good point, the language could be refined to make the possible headers in 104 more clear. Keeping Upload-Offset optional in 104 would be my preference because a `Upload-Offset: 0` looks like no data was saved at all, which might not be true. So I agree with your preference here.